### PR TITLE
Create footer links 'Main,' 'Information For,' 'Topics,' & 'Resources'

### PR DESCRIPTION
**Pull request by smayden**
This ticket creates and styles four of the menus/blocks in the footer and subfooter. 

## Description
This PR adds footer menus/blocks for:
-Main (footer region)
-Information For (subfooter)
-Topics (subfooter)
-Resources (subfooter)

It also has twig templates for styling links in the footer/subfooter regions and for each block (for styling the heading).

## Related Issues
#579 
#580 
#581 
#582 

## How Has This Been Tested?
Tested locally and in Probo

To test:

Add links to each menu: /admin/structure/menu
Link should appear in the sub footer region

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: config/install/system.menu.az-footer-information-for.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+langcode: en
+status: true
+dependencies: {  }
+id: az-footer-information-for
+label: 'Footer - Information For'
+description: 'Generally contains enterprise level navigation links to audience-based sections of Arizona.edu (e.g. current students, faculty).'
```
Comment (joeparsons) [inline config/install/system.menu.az-footer-information-for.yml:6]: Nitpick: institution vs enterprise?
```suggestion
description: 'Generally contains institution level navigation links to audience-based sections of Arizona.edu (e.g. current students, faculty).'
```

Comment (smayden) [reply config/install/system.menu.az-footer-information-for.yml]: Changed the language here

**Comment by smayden** [comment]
Remaining styling:
-Topics menu should wrap to two columns. Looks like we did this with custom css in QS1:
https://bitbucket.org/ua_drupal/ua_quickstart/src/7.x-1.x/modules/custom/uaqs_navigation/uaqs_navigation_global_footer/css/uaqs_navigation_global_footer.css
-Do we want to keep icons for the Resources? How should that be implemented?

Looking ahead:
-Have questions on best implementation for social media (separate module in QS1)

**Comment by smayden** [comment]
Added the split-2-col class to topics menu That Dana had added to Bootstrap ([#276](https://github.com/az-digital/arizona-bootstrap/pull/276)

Can test by adding the css in the inspector:

.split-2-col {
    display: block;
    column-count: 2;
}

image

File: themes/custom/az_barrio/templates/navigation/menu--footer_main.html.twig [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{#
+  We call a macro which calls itself to render the full tree.
+  @see http://twig.sensiolabs.org/doc/tags/macro.html
+#}
+{{ menus.menu_links(items, attributes, 0) }}
+{% macro menu_links(items, attributes, menu_level) %}
+  {% import _self as menus %}
+  {% if items %}
+    {% if menu_level == 0 %}
+      <ul{{ attributes.addClass('nav').addClass('nav-utility').addClass('d-flex').addClass('justify-content-end') }}>
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/navigation/menu--footer_main.html.twig:32]: Might want to consider adding these classes to have this menu stack on mobile.

```suggestion
      <ul{{ attributes.addClass('nav').addClass('nav-utility').addClass('d-flex').addClass('justify-content-end').addClass('flex-column').addClass('flex-md-row') }}>
```

File: themes/custom/az_barrio/templates/navigation/menu--footer_main.html.twig [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          'nav-item',
+          item.is_expanded ? 'menu-item--expanded',
+          item.is_collapsed ? 'menu-item--collapsed',
+          item.in_active_trail ? 'menu-item--active-trail',
+        ]
+      %}
+      <li{{ item.attributes.addClass(classes) }}>
+        {%
+          set link_classes = [
+            'nav-link',
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/navigation/menu--footer_main.html.twig:48]: Add `d-flex` class to have hover underline only the word

```suggestion
            'nav-link d-flex',
```

File: themes/custom/az_barrio/templates/navigation/menu--sub_footer_topics.html.twig [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+{#
+  We call a macro which calls itself to render the full tree.
+  @see http://twig.sensiolabs.org/doc/tags/macro.html
+#}
+{{ menus.menu_links(items, attributes, 0) }}
+{% macro menu_links(items, attributes, menu_level) %}
+  {% import _self as menus %}
+  {% if items %}
+    {% if menu_level == 0 %}
+      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-utility').addClass('split-2-col') }}>
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/navigation/menu--sub_footer_topics.html.twig:32]: I will need to update `split-2-col` in Bootstrap to apply only on medium screens and up

File: themes/custom/az_barrio/templates/blocks/block--az-barrio-footer-menu-info.html.twig [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+{%
+  set classes = [
+    'block',
+    'block-menu',
+    'navigation',
+    'menu--' ~ derivative_plugin_id|clean_class,
+  ]
+%}
+{% set heading_id = attributes.id ~ '-menu'|clean_id %}
+<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-4 col-md-3 col-lg-3')|without('role', 'aria-labelledby') }}>
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/blocks/block--az-barrio-footer-menu-info.html.twig:43]: Shouldn't need the col-lg-3 class
```suggestion
<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-4 col-md-3')|without('role', 'aria-labelledby') }}>
```

File: themes/custom/az_barrio/templates/blocks/block--az-barrio-footer-menu-topics.html.twig [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+{%
+  set classes = [
+    'block',
+    'block-menu',
+    'navigation',
+    'menu--' ~ derivative_plugin_id|clean_class,
+  ]
+%}
+{% set heading_id = attributes.id ~ '-menu'|clean_id %}
+<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-8 col-md-5 col-lg-5')|without('role', 'aria-labelledby') }}>
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/blocks/block--az-barrio-footer-menu-topics.html.twig:43]: Shouldn't need the col-lg-5 class
```suggestion
<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-8 col-md-5')|without('role', 'aria-labelledby') }}>
```

**Review by danahertzberg — APPROVED** [review]

**Comment by smayden** [comment]
I made all of Dana's suggested edits and this PR is ready for review.

**Comment by smayden** [comment]
This ticket is ready for review. FYI Implemented all of Dana's suggestions during the workshop time (thanks @danahertzberg). 

The topics menu has the split-md-2-col class that Dana plans to introduce in Bootstrap. This should make that menu go to two columns are medium and larger displays.

**Review by tadean — APPROVED** [review]

**Review by mmunro-ltrr — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**